pages/[user]/[type].tsx

```diff
@@ -233,72 +234,73 @@ export default function Type(props): Type {
   }
 }
 
-export const getServerSideProps: GetServerSideProps = async (context: GetServerSidePropsContext) => {
-  const dateQuery = context.query?.date ?? null;
-  const date = Array.isArray(dateQuery) && dateQuery.length > 0 ? dateQuery.pop() : dateQuery;
+export const getServerSideProps = async (context: GetServerSidePropsContext) => {
+  // get query params and typecast them to string
+  // (would be even better to assert them instead of typecasting)
+  const userParam = context.query.user as string;
+  const typeParam = context.query.type as string;
+  const dateParam = context.query.date as string | undefined;
 
-  const user = await whereAndSelect(
-    prisma.user.findFirst,
-    {
-      username: context.query.user.toLowerCase(),
+  const user = await prisma.user.findFirst({
+    where: {
+      username: userParam.toLowerCase(),
     },
-    [
-      "id",
-      "username",
-      "name",
-      "email",
-      "bio",
-      "avatar",
-      "startTime",
-      "endTime",
-      "timeZone",
-      "weekStart",
-      "availability",
-      "hideBranding",
-      "theme",
-    ]
-  );
+    select: {
+      id: true,
+      username: true,
+      name: true,
+      email: true,
+      bio: true,
+      avatar: true,
+      startTime: true,
+      endTime: true,
+      timeZone: true,
+      weekStart: true,
+      availability: true,
+      hideBranding: true,
+      theme: true,
+    },
+  });
 
   if (!user) {
     return {
       notFound: true,
-    };
+    } as const;
```

Comment:
Note - for `InferGetServerSidePropsType` to work we have to do `as const` here on the not found response